### PR TITLE
[FIX] Fix linking against protobuff.

### PR DESCRIPTION
**Pull request by ayermolo**
*Description*:
Fix linking against protobuf built as dll.
*Testing*:
Unit tests.
*Documentation*:
NA
[Optional Fixes #issue]

Please see a detailed explanation of how to fill out the fields in the relevant sections in PULL_REQUEST.md.

### CODE REVIEW

**Comment by bertmaher** [comment]
Can you provide some more context here?  Under what conditions does an error occur (and what are the error messages)?  And why does this PR fix it?  Thanks!

File: CMakeLists.txt [hunk 1]
```diff
@@ -10,6 +10,7 @@ option(GLOW_WITH_OPENCL "Build the OpenCL backend" ON)
 option(GLOW_BUILD_EXAMPLES "Build the examples" ON)
 option(GLOW_BUILD_TESTS "Build the tests" ON)
 option(GLOW_WITH_BUNDLES "Build bundles" OFF)
+OPTION(LINK_PROTOBUF_AS_DLL "Link against protobuf build as dynamic libray." OFF)
```
Comment (nadavrot) [inline CMakeLists.txt]: All of the options before this one are spelled with lower case.

File: CMakeLists.txt [hunk 2]
```diff
@@ -66,6 +67,18 @@ if (GLOW_WITH_OPENCL)
...
 
+# sets pre processor flag so some of glow projects
+# can link against protobuf libray that is build as DLL.
+if(LINK_PROTOBUF_AS_DLL)
+  # For protobuf warning when it is build as dll
+  # Supresses a warning that is treated as warning
+  # Basically one of the header files has interface class
+  # containing STL library. Which might cause issues
+  # if things are build with different compilers.
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4251")
```
Comment (nadavrot) [inline CMakeLists.txt]: If we set LINK_PROTOBUF_AS_DLL to True on Linux things would not work. Can we wrap this in an if-def for windows?

File: CMakeLists.txt (hunk 1 above)
Comment (ayermolo) [reply CMakeLists.txt]: Fixed

File: CMakeLists.txt (hunk 2 above)
Comment (ayermolo) [reply CMakeLists.txt]: Added it as an AND.

**Comment by nadavrot** [comment]
LGTM. Unless @bertmaher has any comments I am okay with landing this change as is.

**Pull request merged**